### ISSUE TITLE
Icon shouldnt be required for rules

### ISSUE THREAD

**Issue by psk788** [meshifyiot/carbon-docs#37]
Right now, Icons are required for rules.  Sometimes we may want to evaluate a rule and send a notification but not necessarily trigger an alarm.  Can we make it not required?

**Comment by Ashtonian** [comment]
what is the intention here? removing the required icon would not prevent an "alarm" being triggered. Please explain the use case more fully.

**Comment by willpopski** [comment]
This was used on legacy all the time. need a way to send an alert without changing the status in the UI.

**Comment by MattMadeiro** [comment]
I think the default icon functionality will solve this. Let me know what you think.

Right now, in the dashboard, we display a green circle as the default icon for a node or folder. (This is prior to us fetching all the current alarms and determining the actual icon of the node.) This green circle is hard-coded into the dashboard code, though, and in no way influenced by the icons you've set up in the Admin interface.

We're going to change this. If you create an icon named 'default' in the Admin, we'll use that icon instead of the green circle. We'll also prevent any alarms with the default icon from appearing in the list of alarms at the top of the folder & node list. Likewise, when viewing a single folder or node, in any place where you'd see the alarms for it, we can exclude alarms with the default icon.

When creating a rule, then, you can set it the default icon, and the front-end UI will be unchanged.

**Comment by willpopski** [comment]
That seems like it works fine visually. If we were tracking nodes by status over time, would that fall into the same status as it would with normal green?

**Comment by MattMadeiro** [comment]
Can you clarify how you'd be tracking nodes by status? In the Event History or something?

**Comment by willpopski** [comment]
Event history, or even current event status. It seems like it would show up as an additional filter on the list.

**Comment by MattMadeiro** [comment]
Correct, yeah. You're configuring an alarm when you create the rule, so even if the UI obscures the color/icon change, an alarm is still firing. The node is still entering into that new state.

That said, it's still easy for us—on a per-customer basis, even, in case someone decides they don't like this—to prevent rules with the default icon from showing up in the event history or as a filter in any list.

**Comment by willpopski** [comment]
Maybe a better, general feature, would be to add a checkbox, similar to in legacy, where the status may not need to appear on the dashboard. That way, any icon could be optionally hidden from the dashboard.

Thoughts?

**Comment by decker405** [comment]
Here's how this is supposed to work:

- if icon named "default" does not exist, the Dashboard core uses a green circle as a backup (for nodes with no alarms)
- the dashboard gets all of the nodes and alarms, then computes the icon for each node
- if no alarms exist, it uses the default icon
- based on a Theme option (hideDefaultAlarms=false by default) any alarms with the default icon can hidden

Currently we are missing a few of these items, we need to:

- [ ] create an icon named "default" on Tenant creation with Level 0
- [ ] don't allow editing the name or level of the "default" icon in the UI
- [ ] make sure the "default" icon in the API overrides the Core's backup icon (if it exists)
- [ ] After adding basic Theming, support the use of the `hideDefaultAlarms` property
